precompiles/dispatch-lockdrop/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use core::marker::PhantomData;
+use fp_evm::PrecompileHandle;
+use frame_support::pallet_prelude::IsType;
+use frame_support::{codec::DecodeLimit as _, dispatch::Pays, traits::Get};
+use frame_support::{
+    dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo},
+    traits::ConstU32,
+};
+use frame_system::Config;
+use pallet_evm::GasWeightMapping;
+use pallet_evm_precompile_dispatch::DispatchValidateT;
+use parity_scale_codec::Encode;
+use precompile_utils::prelude::{BoundedBytes, UnboundedBytes};
+use precompile_utils::{keccak256, EvmResult};
+use sp_core::ecdsa::Signature;
+use sp_core::{crypto::AccountId32, H160, H256};
+use sp_core::{ecdsa, U256};
+use sp_io::hashing::keccak_256;
+use sp_runtime::traits::Zero;
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub const LOG_TARGET: &str = "dispatch-lockdrop-precompile";
+
+// ECDSA signature bytes
+type ECDSASignatureBytes = ConstU32<65>;
+
+// `DecodeLimit` specifies the max depth a call can use when decoding, as unbounded depth
+// can be used to overflow the stack.
+// Default value is 8, which is the same as in XCM call decoding.
+pub struct DispatchLockdrop<Runtime, DispatchValidator = (), DecodeLimit = ConstU32<8>>(
+    PhantomData<(Runtime, DispatchValidator, DecodeLimit)>,
+);
+
+#[precompile_utils::precompile]
+impl<Runtime, DispatchValidator, DecodeLimit>
+    DispatchLockdrop<Runtime, DispatchValidator, DecodeLimit>
+where
+    Runtime: pallet_evm::Config + pallet_unified_accounts::Config,
```

Comment:
I'd suggest to replace `pallet_unified_accounts::Config` with a more specific trait dependency.

The functions you need from it should be abstracted behind a trait, `pallet_unified_accounts` should implement that trait, and then this precompile can have a trait bound on that trait instead of the whole pallet.

No need for tight coupling.